holoviews/core/data.py

```diff
@@ -1308,9 +1334,288 @@ def aggregate(cls, columns, kdims, function, **kwargs):
 
 
 
+class GridColumns(DictColumns):
+    """
+    Interface for simple dictionary-based columns format. The dictionary
+    keys correspond to the column (i.e dimension) names and the values
+    are collections representing the values in that column.
+    """
+
+    types = (dict, OrderedDict, cyODict)
+
+    datatype = 'grid'
+
+    @classmethod
+    def reshape(cls, eltype, data, kdims, vdims):
+        if kdims is None:
+            kdims = eltype.kdims
+        if vdims is None:
+            vdims = eltype.vdims
+
+        if not vdims:
+            raise ValueError('GridColumns interface requires at least '
+                             'one value dimension.')
+
+        dimensions = [d.name if isinstance(d, Dimension) else
+                      d for d in kdims + vdims]
+        if isinstance(data, tuple):
+            data = {d: v for d, v in zip(dimensions, data)}
+        elif not isinstance(data, dict):
+            raise ValueError('GridColumns must be instantiated as a '
+                             'dictionary or tuple')
+
+        if 'vdims' in data:
+            vdim_array = data.pop('vdims')
+            for i, vdim in enumerate(vdims):
+                name = vdim.name if isinstance(vdim, Dimension) else vdim
+                data[name] = vdim_array[..., i]
+
+        for dim in kdims+vdims:
+            name = dim.name if isinstance(dim, Dimension) else dim
+            if name not in data:
+                raise ValueError("Values for dimension %s not found" % dim)
+            if not isinstance(data[name], np.ndarray):
+                data[name] = np.array(data[name])
+
+        kdim_names = [d.name if isinstance(d, Dimension) else d for d in kdims]
+        vdim_names = [d.name if isinstance(d, Dimension) else d for d in vdims]
+        expected = [len(data[kd]) for kd in kdim_names]
+        for vdim in vdim_names:
```

Comment:
I am wondering if this bit shouldn't be in ``validate``. Looks to me like ``validate`` is always called when the interface is constructed. Does an exception in ``validate`` trigger a fallback?